modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/introspection/RuntimeSnapshot.java

```diff
@@ -338,19 +338,19 @@ public ServiceInfo getService(String id) {
 
     @Override
     public List<String> getJavaComponentIds() {
-        return javaComponentsIds;
+        return javaComponentsIds.stream()
+                                .sorted()
+                                .collect(Collectors.collectingAndThen(Collectors.toList(),
+                                        Collections::unmodifiableList));
     }
 
     @Override
     public List<String> getXmlComponentIds() {
-        List<String> result = new ArrayList<>();
-
-        for (String cId : getComponentIds()) {
-            if (!javaComponentsIds.contains(cId)) {
-                result.add(cId);
-            }
-        }
-        return result;
+        return getComponentIds().stream()
+                                .filter(cId -> !javaComponentsIds.contains(cId))
```

Comment:
This is not a change request, it can be written in a more functional way:
```suggestion
                                .filter(Predicate.not(javaComponentsIds::contains))
```